cognite_toolkit/cdf_tk/bootstrap.py

```diff
@@ -23,71 +24,209 @@
+    if vars.project is not None and len(vars.project) > 0:
+        if verbose:
+            vars.info += f"  CDF_PROJECT={vars.project} is set correctly.\n"
+    else:
+        vars.error = True
+        vars.info += "  [bold red]ERROR[/]: Environment variable CDF_PROJECT must be set or use --interactive.\n"
+        return vars
+    if vars.audience != f"https://{vars.cluster}.cognitedata.com":
+        vars.warning = True
+        vars.info += f"  [bold yellow]WARNING[/]: IDP_AUDIENCE is set to {vars.audience}, are you sure it shouldn't be https://{vars.cluster}.cognitedata.com?\n"
+    elif verbose:
+        vars.info += f"  IDP_AUDIENCE = {vars.audience} is set correctly.\n"
+    if vars.scopes != f"https://{vars.cluster}.cognitedata.com/.default":
+        vars.warning = True
+        vars.info += f"  [bold yellow]WARNING[/]: IDP_SCOPES is set to {vars.scopes}, are you sure it shouldn't be https://{vars.cluster}.cognitedata.com/.default?\n"
+    elif verbose:
+        vars.info += f"  IDP_SCOPES = {vars.scopes} is set correctly.\n"
+    if interactive:
+        token = False
+        if vars.token is None or len(vars.token) == 0:
+            token = Confirm.ask(
+                "Do you have client id and client secret for a service principal/application? ",
+                choices=["y", "n"],
+            )
+            if not token:
+                vars.token = Prompt.ask("OAuth2 token (CDF_TOKEN)? ", password=True)
+        else:
+            new_token = Prompt.ask("You have set an OAuth2 token (CDF_TOKEN), change it ? ", password=True, default="")
+            if len(new_token) > 0:
+                vars.token = new_token
+            else:
+                print("  Keeping existing token.")
+        if token:
+            azure = Confirm.ask(
+                "Do you have Azure Entra/ActiveDirectory as your identity provider ?", choices=["y", "n"]
+            )
+            name_of_principal = "Service principal/application"
+            if azure:
+                vars.tenant_id = Prompt.ask(
+                    "What is your Entra tenant id (e.g. [italic]12345678-1234-1234-1234-123456789012[/])? ",
+                    default=vars.tenant_id,
+                )
+                name_of_principal = "Application"
+                vars.token_url = f"https://login.microsoftonline.com/{vars.tenant_id}/oauth2/v2.0/token"
+            else:
+                vars.token_url = Prompt.ask(
+                    "What is your identity provider token endpoint (e.g. [italic]https://myidp.com/oauth2/token[/])? "
+                )
+            vars.client_id = Prompt.ask(f"{name_of_principal} client id (CDF_CLIENT_ID)? ", default=vars.client_id)
+            if vars.client_secret is not None and len(vars.client_secret) > 0:
+                new_secret = Prompt.ask(
+                    "You have set a client secret (CDF_CLIENT_SECRET), change it ? ", password=True, default=""
+                )
+                if len(new_secret) > 0:
```

Comment:
This concussed me
```
You have set a client secret (CDF_CLIENT_SECRET), change it ?  ():
```

How should I proceed if I am happy with the secret? Suggest adding (press enter to continue)